update-overlay/create-pr-if-necessary.js

```diff
@@ -0,0 +1,200 @@
+    const sourceJsonContent = JSON.stringify({
+      repo: pluginsRepoUrl,
+      "repo-ref": workspaceCommit,
+      "repo-flat": pluginsRepoFlat === 'true',
+    });
+
+    const workspaceLink = pluginsRepoFlat === 'true' ?
+      `/${pluginsRepoOwner}/${pluginsRepoName}/tree/${workspaceCommit}`
+      : `/${pluginsRepoOwner}/${pluginsRepoName}/tree/${workspaceCommit}/workspaces/${workspaceName}`;
+
+    core.info(`Checking existing content on the target branch`);
+    let needsUpdate = false;
+    try {
+      const checkExistingResponse = await githubClient.repos.getContent({
+        owner: overlayRepoOwner,
+        repo: overlayRepoName,
+        mediaType: {
+          format: 'text'        
+        }, 
+        path: `${workspacePath}/source.json`,
+        ref: overlayRepoBranchName,
+      })
+
+      if (checkExistingResponse.status === 200) {
+        console.log('workspace already exists on the target branch');
+        const data = checkExistingResponse.data;
+        if ('content' in data && data.content !== undefined) {
+          const content = Buffer.from(data.content, 'base64').toString();
+          const sourceInfo = JSON.parse(content); 
+          if (sourceInfo['repo-ref'] === workspaceCommit.trim() &&
+              sourceInfo['repo'] === pluginsRepoUrl &&
+              sourceInfo['flat'] === (pluginsRepoFlat === 'true')
+            ) {
+            core.notice(
+              `Workspace ${workspaceName} already exists on branch ${overlayRepoBranchName} with the same commit ${workspaceCommit.substring(0,7)}`,
+              { title: 'Workspace skipped' }
+            )
+            return;
+          }
+        }
+        needsUpdate = true;
+      }
+    } catch(e) {
+      if (e instanceof Object && 'status' in e && e.status === 404) {
+        console.log(`workspace ${workspaceName} not found on branch ${overlayRepoBranchName}`)
+      } else {
+        throw e;
+      }
+    }
+
+    core.info(`Checking pull request existence`);
+    const existingPRs = await githubClient.pulls.list({
+      owner: overlayRepoOwner,
+      repo: overlayRepoName,
+      base: overlayRepoBranchName,
+      head: `${overlayRepoOwner}:${targetPRBranchName}`
+    })
+
+    if (existingPRs.status === 200 && existingPRs.data.length === 1) {
+      core.notice(
+        `Pull request for workspace ${workspaceName} based on branch ${targetPRBranchName} already exists; do not try to create it again.`,
+        { title: 'Workspace skipped' }
+      )
+      return;
+    }
+
+    core.info(`Checking pull request branch existence`);
+    let prBranchExists = false;
+    try {
+      const prCheckResponse = await githubClient.git.getRef({
+        owner: overlayRepoOwner,
+        repo: overlayRepoName,
+        ref: `heads/${targetPRBranchName}`
+      })
+
+      if (prCheckResponse.status === 200) {
+        console.log('pull request branch already exists, but the corresponding PR is missing. Only the PR will be created.')
```

Comment:
should these be using `console.log` too?

Reply:
I guess I can change them to core.info() ...

Reply:
Fixed